packages/fx-core/src/plugins/resource/simpleauth/webAppClient.ts

```diff
@@ -46,13 +50,31 @@ export class WebAppClient {
       this.credentials,
       this.subscriptionId
     );
+    this.resourceManagementClient = new Providers(
+      new ResourceManagementClientContext(this.credentials, this.subscriptionId)
+    );
     this.ctx = ctx;
   }
 
-  public async createWebApp(): Promise<{ endpoint: string, skuName: string }> {
+  public async createWebApp(): Promise<{ endpoint: string; skuName: string }> {
     let skuName: string;
 
     try {
+      // Check and register resource provider
+      try {
+        DialogUtils.progressBar?.next(Constants.ProgressBar.provision.registerResourceProvider);
+        await ResourceManagementClient.ensureResourceProviders(
+          this.resourceManagementClient,
+          Constants.RequiredResourceProviders
+        );
+      } catch (error) {
+        throw ResultFactory.UserError(
```

Comment:
error will block the following steps?

Reply:
Yes. As user does not have permission to register the RP, we think we should block provision and ask user to manually register the RP.

Reply:
The scenario is the RP is already registered although user don't have the permission to register. And the user has permission to provision.

Reply:
Here we will check whether the RP is registered. If registered, we will not register again. So in this scenario, it will not throw error here.

Reply:
If the user don't have permission to check whether registered?

Reply:
Seems we can not find a permission on checking the register status.
![image](https://user-images.githubusercontent.com/63089166/127112895-b47bd1a4-7503-4eab-8d5e-e19cd2bb26d8.png)

Shall we separate the check and registering, and only throw error when fail to register?

Reply:
I think register the RP is just tries to do something. If cannot sucessfully run, it should not block provision.

Reply:
Do you mean we just tries to register the RP instead of checking the status first, and no matter whether it success, we will go on and try to provision app services?

Reply:
Fixed. Thanks!